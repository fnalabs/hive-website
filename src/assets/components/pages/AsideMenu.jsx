import React from 'react'
import { NavLink } from 'react-router-dom'

import { AnalyticsLink } from 'common'

const AsideMenu = () => (
  <aside className='menu'>
    <p className='menu-label has-text-weight-bold'>v2.1.3</p>
    <ul className='menu-list'>
      <li><NavLink to='/overview' activeClassName='is-active'>Overview</NavLink>
        <ul>
          <li><NavLink to='/model' activeClassName='is-active'>Data Model</NavLink></li>
          <li><NavLink to='/domain' activeClassName='is-active'>Domain Logic</NavLink></li>
          <li><NavLink to='/infrastructure' activeClassName='is-active'>Infrastructure</NavLink></li>
          <li><NavLink to='/telemetry' activeClassName='is-active'>Telemetry</NavLink></li>
        </ul>
      </li>

      <li><NavLink to='/start' activeClassName='is-active'>Get Started</NavLink>
        <ul>
          <li><NavLink to='/setup' activeClassName='is-active'>Setup</NavLink></li>
          <li className='menu-label has-text-weight-bold'>Examples:</li>
          <li><NavLink to='/basic' activeClassName='is-active'>Basic</NavLink></li>
          <li><NavLink to='/rest' activeClassName='is-active'>Rest</NavLink></li>
          <li><NavLink to='/cqrs-es' activeClassName='is-active'>CQRS/ES</NavLink></li>
        </ul>
      </li>

      <li><NavLink to='/documentation' activeClassName='is-active'>Documentation</NavLink>
        <ul>
          <li><NavLink to='/environments' activeClassName='is-active'>Environments</NavLink></li>
          <li><AnalyticsLink to='https://fnalabs.github.io/hive-io/' target='_blank' rel='noopener noreferrer'>API</AnalyticsLink></li>
        </ul>
      </li>
    </ul>
  </aside>
)

export default AsideMenu
