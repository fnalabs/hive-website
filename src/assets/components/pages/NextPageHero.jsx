import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { AnalyticsLink } from 'common'
import { Hero, HeroBody } from 'common/hero'
import { ArrowRight } from 'icons'

const toRegex = /^https:\/\//

export default class NextPageHero extends Component {
  static propTypes = {
    to: PropTypes.string.isRequired
  }

  render () {
    const to = this.props.to

    return (
      <Hero size='small' className='is-hive'>
        <HeroBody centered>
          {toRegex.test(to)
            ? <AnalyticsLink to={to}><ArrowRight height='48' /></AnalyticsLink>
            : <Link to={to}><ArrowRight height='48' /></Link>}
        </HeroBody>
      </Hero>
    )
  }
}
