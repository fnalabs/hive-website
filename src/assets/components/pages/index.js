// site's main pages
export { default as Home } from './home/Home.jsx'
export { default as Overview } from './overview/Overview.jsx'
export { default as Start } from './start/Start.jsx'

// policy pages
export { Cookie } from './policies/Cookie.jsx'
export { Privacy } from './policies/Privacy.jsx'
