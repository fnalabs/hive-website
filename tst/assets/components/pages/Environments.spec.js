/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { MemoryRouter } from 'react-router'
import ReactGA from 'react-ga'

import { Environments } from 'pages'

jest.mock('react-ga')

describe('<Environments />', () => {
  it('should render Environments page static content', () => {
    const tree = renderer.create(<MemoryRouter><Environments location={{ pathname: '/environments' }} /></MemoryRouter>).toJSON()

    expect(tree).toMatchSnapshot()
    expect(ReactGA.pageview).not.toBeCalled()
  })

  it('should render Environments page static content and report page view', () => {
    shallow(<Environments location={{ pathname: '/environments' }} />, {
      context: { isConsent: true }
    })

    expect(ReactGA.pageview).toBeCalledTimes(1)
  })
})
