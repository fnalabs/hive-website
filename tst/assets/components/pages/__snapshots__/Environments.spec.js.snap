// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Environments /> should render Environments page static content 1`] = `
Array [
  <article
    className="section is-medium"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <header
          className="column is-narrow is-hidden-touch"
        >
          <aside
            className="menu"
          >
            <p
              className="menu-label has-text-weight-bold"
            >
              v2.1.0
            </p>
            <ul
              className="menu-list"
            >
              <li>
                <a
                  aria-current={null}
                  href="/overview"
                  onClick={[Function]}
                >
                  Overview
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/model"
                      onClick={[Function]}
                    >
                      Data Model
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/domain"
                      onClick={[Function]}
                    >
                      Domain Logic
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/infrastructure"
                      onClick={[Function]}
                    >
                      Infrastructure
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/telemetry"
                      onClick={[Function]}
                    >
                      Telemetry
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/start"
                  onClick={[Function]}
                >
                  Get Started
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/setup"
                      onClick={[Function]}
                    >
                      Setup
                    </a>
                  </li>
                  <li
                    className="menu-label has-text-weight-bold"
                  >
                    Examples:
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/basic"
                      onClick={[Function]}
                    >
                      Basic
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/rest"
                      onClick={[Function]}
                    >
                      Rest
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/cqrs-es"
                      onClick={[Function]}
                    >
                      CQRS/ES
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/documentation"
                  onClick={[Function]}
                >
                  Documentation
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/environments"
                      onClick={[Function]}
                    >
                      Environments
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-io/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      API
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </aside>
        </header>
        <section
          className="column content"
          role="document"
        >
          <h1>
            Environments
          </h1>
          <p>
            The tables below contain all of the environment variables used to configure each microservice container. Default values are defined where applicable as well as descriptions for each.
          </p>
          <h2
            id="shared"
          >
            Shared
          </h2>
          <p>
            The following envrionment variables are shared across all microservice images:
          </p>
          <div
            className="table-container"
          >
            <table
              className="table is-striped is-hoverable"
            >
              <thead>
                <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Type
                  </th>
                  <th>
                    Default
                  </th>
                  <th>
                    Description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    NODE_ENV
                  </td>
                  <td>
                    String
                  </td>
                  <td>
                    'production'
                  </td>
                  <td>
                    microservice runtime environment
                  </td>
                </tr>
                <tr>
                  <td>
                    PORT
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    3000
                  </td>
                  <td>
                    microservice port to listen on
                  </td>
                </tr>
                <tr>
                  <td>
                    HTTP_VERSION
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    2
                  </td>
                  <td>
                    HTTP version for backward compatibility
                  </td>
                </tr>
                <tr>
                  <td>
                    SECURE
                  </td>
                  <td>
                    String
                  </td>
                  <td>
                    'false'
                  </td>
                  <td>
                    whether to run microservice secure or not. defaults to 'false' since we cannot provide certificates
                  </td>
                </tr>
                <tr>
                  <td>
                    CLUSTER_SIZE
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    [total CPUs available]
                  </td>
                  <td>
                    defaults to the total available CPUs allocated to the container or to the size you specify here
                  </td>
                </tr>
                <tr>
                  <td>
                    SSL_CERT
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    default path for SSL certificate file or the full certificate
                  </td>
                </tr>
                <tr>
                  <td>
                    SSL_KEY
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    default path for SSL key file or the full key
                  </td>
                </tr>
                <tr>
                  <td>
                    PING_URL
                  </td>
                  <td>
                    String
                  </td>
                  <td>
                    '/ping'
                  </td>
                  <td>
                    URL to use for shallow health checks for the microservice
                  </td>
                </tr>
                <tr>
                  <td>
                    ACTOR
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    Actor (Model) the microservice is responsible for
                  </td>
                </tr>
                <tr>
                  <td>
                    ACTOR_LIB
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    module where the ACTOR resides
                  </td>
                </tr>
                <tr>
                  <td>
                    ACTOR_URLS
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    comma-separated URLs associated with the Actor
                  </td>
                </tr>
                <tr>
                  <td>
                    TELEMETRY
                  </td>
                  <td>
                    String
                  </td>
                  <td>
                    'false'
                  </td>
                  <td>
                    whether to run OpenTelemetry integration
                  </td>
                </tr>
                <tr>
                  <td>
                    TELEMETRY_PLUGINS
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    JSON string of OpenTelemetry plugins to enable
                  </td>
                </tr>
                <tr>
                  <td>
                    TELEMETRY_SERVICE_NAME
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    service name for OpenTelemetry
                  </td>
                </tr>
                <tr>
                  <td>
                    TELEMETRY_URL_METRICS
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    OpenTelemetry Collector URL for metrics
                  </td>
                </tr>
                <tr>
                  <td>
                    TELEMETRY_URL_TRACES
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    OpenTelemetry Collector URL for traces
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <h2>
            Base
          </h2>
          <p>
            All of the environment variables in 
            <a
              href="#shared"
            >
              Shared
            </a>
            .
          </p>
          <h2>
            Producer
          </h2>
          <p>
            All of the environment variables in 
            <a
              href="#shared"
            >
              Shared
            </a>
             and the following:
          </p>
          <div
            className="table-container"
          >
            <table
              className="table is-striped is-hoverable"
            >
              <thead>
                <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Type
                  </th>
                  <th>
                    Default
                  </th>
                  <th>
                    Description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    EVENT_STORE_TOPIC
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    Kafka topic the events will be stored under
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_ID
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    unique identifier for Kafka client connection
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_BROKERS
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    comma separated URLs where Kafka is hosted
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_BUFFER
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    100
                  </td>
                  <td>
                    maximum number of incoming messages to batch
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_TIMEOUT
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    2000
                  </td>
                  <td>
                    time (in \`ms\`) to poll Kafka for delivery reports
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <h2>
            Consumer
          </h2>
          <p>
            All of the environment variables in 
            <a
              href="#shared"
            >
              Shared
            </a>
             and the following:
          </p>
          <div
            className="table-container"
          >
            <table
              className="table is-striped is-hoverable"
            >
              <thead>
                <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Type
                  </th>
                  <th>
                    Default
                  </th>
                  <th>
                    Description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    EVENT_STORE_TOPIC
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    Kafka topic the events will be consumed from
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_ID
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    unique identifier for Kafka client connection
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_GROUP_ID
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    defines Kafka Consumer group id
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_BROKERS
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    comma separated URLs where Kafka is hosted
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_FROM_START
                  </td>
                  <td>
                    String
                  </td>
                  <td>
                    'false'
                  </td>
                  <td>
                    tells Consumer whether or not to start at the beginning of the topic
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_PARTITIONS
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    1
                  </td>
                  <td>
                    tells Consumer how many partitions to consume
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_BUFFER
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    null
                  </td>
                  <td>
                    maximum number of incoming messages to batch
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_TIMEOUT
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    null
                  </td>
                  <td>
                    time (in \`ms\`) to poll Kafka for delivery reports
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <h2>
            Stream Processor
          </h2>
          <p>
            All of the environment variables in 
            <a
              href="#shared"
            >
              Shared
            </a>
             and the following:
          </p>
          <div
            className="table-container"
          >
            <table
              className="table is-striped is-hoverable"
            >
              <thead>
                <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Type
                  </th>
                  <th>
                    Default
                  </th>
                  <th>
                    Description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    PROCESSOR_TYPE
                  </td>
                  <td>
                    String
                  </td>
                  <td>
                    'producer'
                  </td>
                  <td>
                    type of Stream Processor microservice you wish to run (can also be 'consumer' or 'stream_processor')
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_PRODUCER_TOPIC
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    Kafka topic the events will be stored under
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_CONSUMER_TOPIC
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    Kafka topic the events will be consumed from
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_ID
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    unique identifier for Kafka client connection
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_GROUP_ID
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    defines Kafka Consumer group id
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_BROKERS
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    comma separated URLs where Kafka is hosted
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_FROM_START
                  </td>
                  <td>
                    String
                  </td>
                  <td>
                    'false'
                  </td>
                  <td>
                    tells Consumer whether or not to start at the beginning of the topic
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_PARTITIONS
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    1
                  </td>
                  <td>
                    tells Consumer how many partitions to consume
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_BUFFER
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    100
                  </td>
                  <td>
                    maximum number of incoming messages to batch
                  </td>
                </tr>
                <tr>
                  <td>
                    EVENT_STORE_TIMEOUT
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    2000
                  </td>
                  <td>
                    time (in \`ms\`) to poll Kafka for delivery reports
                  </td>
                </tr>
                <tr>
                  <td>
                    CACHE_URL
                  </td>
                  <td>
                    String
                  </td>
                  <td />
                  <td>
                    URL where Redis is hosted
                  </td>
                </tr>
                <tr>
                  <td>
                    LOCK_TTL
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    2000
                  </td>
                  <td>
                    Redlock time to live before lock is released
                  </td>
                </tr>
                <tr>
                  <td>
                    LOCK_DRIFT_FACTOR
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    0.01
                  </td>
                  <td>
                    Redlock drift factor setting
                  </td>
                </tr>
                <tr>
                  <td>
                    LOCK_RETRY_COUNT
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    0
                  </td>
                  <td>
                    Redlock retry count setting, should be set to zero for concurrency
                  </td>
                </tr>
                <tr>
                  <td>
                    LOCK_RETRY_DELAY
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    400
                  </td>
                  <td>
                    Redlock retry delay in milliseconds
                  </td>
                </tr>
                <tr>
                  <td>
                    LOCK_RETRY_JITTER
                  </td>
                  <td>
                    Number
                  </td>
                  <td>
                    400
                  </td>
                  <td>
                    Redlock random retry jitter in milliseconds to randomize retries
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </section>
      </div>
    </div>
  </article>,
  <section
    className="hero is-small is-hive"
  >
    <div
      className="hero-body"
    >
      <nav
        className="level is-mobile"
      >
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="previous page"
            href="/documentation"
            onClick={[Function]}
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                />
              </svg>
            </span>
          </a>
        </div>
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="next page"
            href="https://fnalabs.github.io/hive-io/"
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                />
              </svg>
            </span>
          </a>
        </div>
      </nav>
    </div>
  </section>,
]
`;
